src/main/java/net/fabricmc/tinyremapper/TinyRemapper.java

```diff
@@ -397,18 +384,46 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
 		List<ClassInstance> ret = new ArrayList<ClassInstance>();
 
 		if (file.toString().endsWith(".class")) {
-			ClassInstance res = analyze(isInput, tags, srcPath, Files.readAllBytes(file));
+			ClassInstance res = analyze(isInput, tags, srcPath, OptionalInt.empty(), Files.readAllBytes(file)); // FIXME?
 			if (res != null) ret.add(res);
 		} else {
 			URI uri = new URI("jar:"+file.toUri().toString());
 			FileSystem fs = FileSystemHandler.open(uri);
 			fsToClose.add(fs);
-
+			
+			boolean isMrj = false;
+			if(file.getFileName().toString().endsWith(".jar")){
+				System.out.println("jar");
```

Comment:
I recall java util's jar file api has support for getting version-aware jar entries. If there is existing api, we should use those than build our own since our own is vulnerable to ignoring impl/api requirements